build/Microsoft.DotNet.Cli.Prepare.targets

```diff
@@ -95,4 +100,18 @@
     <!-- SetTelemetryProfile -->
     <SetEnvVar Name="DOTNET_CLI_TELEMETRY_PROFILE" Value="https://github.com/dotnet/cli;$(CommitHash)" />
   </Target>
+
+  <Target DependsOnTargets="Init" Name="RestorePackages">
+
+    <!-- Clean NuGet Temp Cache on Linux (seeing some issues on Linux) -->
+    <!-- TODO: do we still need this?  Seems unnecessary -->
+    <RemoveDir
+      Condition=" '$(OSName)' != 'win' AND '$(OSName)' != 'osx' "
+      Directories="/tmp/NuGet" />
+
+    <DotNetRestore WorkingDirectory="$(RepoRoot)/src" ToolPath="$(DotNetPath)" />
+    <DotNetRestore WorkingDirectory="$(RepoRoot)/tools" ToolPath="$(DotNetPath)" />
+
```

Comment:
Ok, lgtm. I want to be sure we don't start incorporating features just because they exist. Offline conversation showed some nice immediate logging wins. I do want to be sure that the Params we expose on these tasks match their cmd line equivalents so translation is easy.